tools/apidiff/Makefile

```diff
@@ -157,32 +157,30 @@ endif
 # easy-to-type helper targets.
 # one rule to create all the api diffs
 
-all-local:: $(APIDIFF_DIR)/api-diff.html
+all-local:: bundle.zip $(APIDIFF_DIR)/api-diff.html
 
 # Rules to re-create the reference infos. The rules target the updated-references
 # directory, so that they're only invoked manually (otherwise the reference infos
 # would be updated every time the assemblies change, which is not what we want).
 
+bundle.zip:
+	curl $(APIDIFF_REFERENCES) > bundle.zip
```

Comment:
A couple of issues:

* The *.xml files don't depend on bundle.zip, so it won't necessarily be downloaded before running mono-api-info.
* `bundle.zip` won't be re-downloaded after changing the url.
* If a previous download is incomplete (interrupted by ctrl-c for instance), it won't be automatically detected and re-downloaded.
* We can't change the url to a redirect, because curl won't follow it.

So, suggested improvements:

```make
# split the URL in words based on the path separator, and then chose the 6th word (the hash) in the bundle zip filename
BUNDLE_ZIP=bundle-$(word 6,$(subst /, ,$(APIDIFF_REFERENCES))).zip
$(BUNDLE_ZIP):
        # download to a temporary filename so interrupted downloads won't prevent re-downloads.
	$(Q_GEN) curl -L $(APIDIFF_REFERENCES) > $@.tmp
        $(Q) mv $@.tmp $@

.unzip.stamp: $(BUNDLE_ZIP)
    $(Q) rm -Rf temp
    $(Q_GEN) unzip -d temp $(BUNDLE_ZIP)
    $(Q) touch $@

# the semi-colon at the end means an empty recipe, and is required for make to consider pattern rules
temp/%.dll: .unzip.stamp ;
```